libs/cri-api-service/src/main/java/uk/gov/di/ipv/core/library/criapiservice/CriApiService.java

```diff
@@ -358,4 +384,37 @@ private VerifiableCredentialResponse getVerifiableCredentialResponseForApplicati
         }
         return vcResponseBuilder.build();
     }
+
+    @Tracing
+    public Optional<JWK> getKeyFromJwksEndpoint(URI jwksEndpoint) {
```

Comment:
One thing to consider - if you pull the other logic out into an `OAuthKeyService`, should this move there as well?

In the next ticket we're going to call the JWKS endpoint from orchestration, rather than a CRI - which will have an almost identical implementation.